### PR TITLE
:seedling: Bump dependencies, primarily CoreDNS for Kubernetes v1.23

### PR DESCRIPTION
**Pull request by randomvariable**
Signed-off-by: Naadir Jeewa <jeewan@vmware.com>

**What this PR does / why we need it**:
Kubernetes v1.23 will ship with CoreDNS 1.8.6, which means a new version of corefile-migration as per https://github.com/kubernetes/kubernetes/pull/106091 . Whilst I was here, bump all the other dependencies.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by sbueringer** [comment]
Not sure if we want to keep the k/k dependencies sync with the one our CR version is using (even on the patch-level).

File: go.mod [hunk 1]
```diff
@@ -5,38 +5,38 @@ go 1.16
...
-	google.golang.org/grpc v1.40.0
-	k8s.io/api v0.22.2
-	k8s.io/apiextensions-apiserver v0.22.2
-	k8s.io/apimachinery v0.22.2
-	k8s.io/apiserver v0.22.2
-	k8s.io/client-go v0.22.2
-	k8s.io/cluster-bootstrap v0.22.2
-	k8s.io/component-base v0.22.2
+	go.etcd.io/etcd/api/v3 v3.5.1
+	go.etcd.io/etcd/client/v3 v3.5.1
```
Comment (fabriziopandini) [inline go.mod:28]: FYI then PR for etcd 3.5.1 is not yet merged in k/k, not sure we want to bump before this

File: go.mod [hunk 2]
```diff
@@ -5,38 +5,38 @@ go 1.16
...
+	golang.org/x/oauth2 v0.0.0-20211104180415-d3ed0bb246c8
+	google.golang.org/grpc v1.42.0
+	k8s.io/api v0.22.3
+	k8s.io/apiextensions-apiserver v0.22.3
+	k8s.io/apimachinery v0.22.3
+	k8s.io/apiserver v0.22.3
+	k8s.io/client-go v0.22.3
+	k8s.io/cluster-bootstrap v0.22.3
+	k8s.io/component-base v0.22.3
```
Comment (fabriziopandini) [inline go.mod:37]: Usually we are bumping this in sync with controller runtime, but no strong opinions for a patch version bump

Comment (randomvariable) [reply go.mod:37]: Oh, ok.

File: go.mod (hunk 1 above)
Comment (randomvariable) [reply go.mod:28]: Has it not, it's definitely going in though, so maybe i'll just put a 
/hold

**Comment by randomvariable** [comment]
/unhold

I've removed the etcd bump as it's still baking in https://github.com/kubernetes/kubernetes/pull/105706

File: go.mod [hunk 3]
```diff
@@ -5,29 +5,29 @@ go 1.16
...
+	github.com/gobuffalo/flect v0.2.4
 	github.com/google/go-cmp v0.5.6
 	github.com/google/go-github/v33 v33.0.0
 	github.com/google/gofuzz v1.2.0
 	github.com/gosuri/uitable v0.0.4
 	github.com/mattn/go-runewidth v0.0.13 // indirect
-	github.com/onsi/ginkgo v1.16.4
-	github.com/onsi/gomega v1.16.0
+	github.com/onsi/ginkgo v1.16.5
+	github.com/onsi/gomega v1.17.0
```
Comment (vincepri) [inline go.mod:22]: These should match controller runtime

Comment (sbueringer) [reply go.mod:22]: Just out of curiosity, why? Do we have dependencies or is it just a matter of consistency?

Comment (sbueringer) [reply go.mod:22]: Good point about pulling multiple versions

Comment (randomvariable) [reply go.mod:22]: Sure, no probs.

Comment (sbueringer) [reply go.mod:22]: @randomvariable PR has been merged

**Review by vincepri — CHANGES_REQUESTED** [review]
@randomvariable Can you rebase?

**Comment by sbueringer** [comment]
I'll take it up in #5761 (co-authored the commit)

So we can close this PR once #5761 has been merged.

**Pull request closed without merging**